go-controller/pkg/ovn/gateway_init.go

```diff
@@ -282,26 +282,27 @@ func gatewayInit(nodeName string, clusterIPSubnet []*net.IPNet, hostSubnets []*n
 		}
 	}
 
-	// Default SNAT rules.
-	externalIPs := make([]net.IP, len(l3GatewayConfig.IPAddresses))
-	for i, ip := range l3GatewayConfig.IPAddresses {
-		externalIPs[i] = ip.IP
-	}
-	for _, entry := range clusterIPSubnet {
-		externalIP, err := gatewayForSubnet(externalIPs, entry)
-		if err != nil {
-			return fmt.Errorf("failed to create default SNAT rules for gateway router %s: %v",
-				gatewayRouter, err)
+	if !l3GatewayConfig.DisableSNATExternalGWs {
+		// Default SNAT rules.
```

Comment:
doesn't this mean that if you set DisableSNATExternalGWs it will disable SNAT for _all_ namespaces, not just namespaces with the annotation?

Reply:
yeah, the idea is to not SNAT if that option is set and then in addLogicalPort/deleteLogicalPort add/delete NAT rules if the pods do not have the annotation to go thru external route

Reply:
can you add a comment here pointing to where "default" SNAT rules get set up for non-external-gws namespaces?

Reply:
ack

Reply:
done